rest/method_item_patch.go

```diff
@@ -64,6 +125,11 @@ func itemPatch(ctx context.Context, r *http.Request, route *RouteMatch) (status
 		e = NewError(err)
 		return e.Code, nil, e
 	}
+
+	if isNoContent(r) {
+		return 204, nil, nil
+	}
+
```

Comment:
To include an E-tag in the response, an item must be returned:
https://github.com/rs/rest-layer/blob/48c11e50091e8c988780853a3aaa8af25cf98d42/rest/response.go#L74

Perhaps you can instead replace:
https://github.com/rs/rest-layer/blob/48c11e50091e8c988780853a3aaa8af25cf98d42/rest/handler.go#L51

with:
```go
skipBody := isNoContent(r)
```

Require you to handle the `r.Method == "HEAD"` as well in the `isNoContent` implementation. This header checks would now work for all update requests. Should also probably ignore the headers for `r.Method == "GET"`.

Reply:
Oh yes, totally forgot about ETags. I will add at least one check for PATCH to check about the ETag in the response. Also I guess I will need to add some tests for other HTTP methods regarding `skipBody`. Thank @smyrman good catch.

Reply:
`HEAD` requests are allowed to contain a body:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

> A response to a HEAD method should not have a body. If so, it must be ignored. Even so, entity headers describing the content of the body, like Content-Length may be included in the response. They don't relate to the body of the HEAD response, which should be empty, but to the body of similar request using the GET method would have returned as a response.

However, failed PATCH,PUT or PAST requests should include an error. Maybe we need two parameters?

```go
skipError := r.Method == "HEAD"
skipOKBody := isNoContent(r)
```

Reply:
But what if `HEAD` has an error. Shouldn't we get the formatted error with appropriate `message` generated by `rest-layer`.